### ISSUE TITLE
use of case in our context file

### ISSUE THREAD

**Issue by cboettig** [codemeta/codemeta#132]
We usually define our terms to match those used in schema.org, e.g. `email`,  `schema:email`, `name`, `schema:name`, but for some reason we have gone to all lowercase for two types: we have `person` and `organization` while schema.org has `schema:Person` and `schema:Organization`.  This matters: e.g. I came across some errors in one example, where the codemeta file refers to `@type: Person`, (https://github.com/codemeta/codemeta/blob/master/examples/example-codemeta-full.json#L67) which isn't a recognized type!  It has to be either `person` (as defined by our context) or refer directly to the `schema` type, `schema:Person`.  (E.g. resolving this all to RDF results in a missing/default namespace for 'Person')

I think this should be considered an error in our `codemeta.jsonld` context, which should really be using capital `Person` and capital `Organization`, but wanted to double check with @mbjones @gothub before I start editing our (already released) `codemeta.jsonld`.  (Can probably fix #129 at the same time, and if there are no other obvious issues, plan for a new release sometime before we submit the manuscript?)

**Comment by mbjones** [comment]
Good to discuss this.   I thought we had decided to use lowercase throughout for consistency, even if the schema.org term or other vocabulary does something different.

Note that we have not yet actually released codemeta 1.0, because we have not yet cut the tag for it.  It is on a branch, awaiting final edits (in particular, awaiting the namespace DOI resolution issue #125 to get worked out.

**Comment by cboettig** [comment]
Yeah, I could go either way.  I imagine `schema.org` had a good reason for this convention, though it's not obvious to me.  I guess either way risks some audience making an error, as our own examples illustrate.  When in doubt though I tend to want to default to what some well-established example does rather than strike out on my own; that way I can always pass the buck.  

Oh, hmm, I must be confused, but looks like we have a 1.0 tag already(?) https://github.com/codemeta/codemeta/releases/tag/1.0. Sounds like @mfenner should have #125 fixed in the near future, so maybe a good time to finalize these schema edits before we cut the tag.

**Comment by gothub** [comment]
I agree with using all lowercase for our terms, the user shouldn't have to know from which schema
they come from, regardless of where they resolve to. 

Using '@type: Person' in the example file is an an error in the
example file, as our context doesn't define a default schema. I'll change this example to '@type: person'. 

The terms 'person' and 'organization' aren't in the crosswalk, as they are used as types to distinguish instances of 'agent'. Should they be promoted to 'concepts'? 

BTW, I originally put the following terms in the context, so we could keep the lowercase convention:

```
      "person":{
         "@id":"schema:Person"
      },
      "organization":{
         "@id":"schema:Organization"
      }
```

**Comment by cboettig** [comment]
@gothub right, I'm glad we explicitly include them in our context, like you say a user should not have to keep track of schemas in their `codemeta.json`.  I just worry that we'll have some users who might write a lot of json-ld and thus be super familiar with schema.org's conventions and have `Person` hardwired in their brain instead of `person`.  Such a user could easily take a quick glance at our context file and think "good, this context matches up with schema.org` and thus make the same typo you made in the examples.  

Regarding the crosswalk, yes, that's a good point.  I think they should be put into the crosswalk (I also think we want schema.org as a column in the crosswalk (#103), even though we've already implicitly cross-walked it by writing out the context file).

**Comment by gothub** [comment]
The 1.0 should maybe not have been defined. Do we keep the 1.0 tag until  [#125](https://github.com/codemeta/codemeta/issues/125)) is fixed? It looks like it is possible to redefine a tag, i.e. to "0.9", if desired.

**Comment by cboettig** [comment]
re-defining the tag sounds good to me, I don't see that breaking anything important.

**Comment by mbjones** [comment]
Hmmm.. my preference is to never rewrite tags, as anyone with a copy will require a force merge to get rid of it.  I think we need to version is to 1.0.1, but not until we are *actually* ready to release.

**Comment by cboettig** [comment]
ok

**Comment by mfenner** [comment]
See my comment a moment ago regarding #125. It might be fixed as early as tomorrow, because an updated DOI content resolver might not be needed after all.

Regarding lowercase/uppercase in schema.org: classes are uppercase and properties lowercase, e.g. `SoftwareSourceCode` or `Person` for the former and e.g. `datePublished` for the latter.

**Comment by cboettig** [comment]
Thanks @mfenner , that makes sense.  So is this something we want to follow as well, or best we just stick with both classes and properties as starting with a lower-case?

**Comment by mfenner** [comment]
I personally have no strong opinion. Depends a little bit on how much you want to align with schema.org.

**Comment by cboettig** [comment]
Closing this in favor of #134.  The current proposed format, codemeta-v2.jsonld adopts the schema.org convention.  

Personally I think it's very helpful to have an easy way to distinguish classes/types from properties. 

 Having the convention of capitalization makes it clear what things are properties (and can thus be used on the left-hand-side in the json) and what things are *not*, and should only appear as an argument to `@type`.  

 It can also serve as a useful reminder to explicitly include `@type` when the property of interest (say, `author`) has a specific node type (like `Person`), which is considered best practice.  For instance, we say `programmingLanguage` is `schema:programmingLanguage` property, which can have either the basic data type `Text`, or be a node of type `ComputerLanguage` (http://schema.org/SoftwareSourceCode).  So our example https://github.com/codemeta/codemeta/blob/master/examples/example-codemeta-full.json#L75-L79 might do better to include a `"@type": "ComputerLanguage"` line, to be explicit about what this node is, just like we already do for `Person` and `Organization` nodes.
